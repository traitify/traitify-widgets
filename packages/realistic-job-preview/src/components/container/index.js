import PropTypes from "prop-types";
import Context from "lib/context";
import ErrorBoundary from "./error-boundary";
import Theme from "./theme";
import useReducer from "./use-reducer";

function Container({children, className = null, ...props}) {
  const state = useReducer({...props});

  return (
    <Context.Provider value={state}>
      <Theme className={className}>
        <ErrorBoundary>{children}</ErrorBoundary>
      </Theme>
    </Context.Provider>
  );
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};

export default Container;
