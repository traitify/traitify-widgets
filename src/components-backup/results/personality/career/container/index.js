import PropTypes from "prop-types";
import TraitifyPropTypes from "lib/helpers/prop-types";
import useDidMount from "lib/hooks/use-did-mount";
import useDidUpdate from "lib/hooks/use-did-update";
import withTraitify from "lib/with-traitify";
import CareerFilter from "../filter";
import CareerList from "../list";
import CareerModal from "../modal";
import style from "./style.scss";

function CareerContainer(props) {
  const {ui, isReady, setElement} = props;
  const state = {};

  useDidMount(() => { ui.trigger("Careers.initialized", {props, state}); });
  useDidUpdate(() => { ui.trigger("Careers.updated", {props, state}); });

  return (
    isReady("results") && (
      <div className={style.container} ref={setElement}>
        <CareerFilter {...props} />
        <CareerList {...props} />
        <CareerModal {...props} />
      </div>
    )
  );
}
CareerContainer.propTypes = {
  isReady: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired,
  setElement: PropTypes.func.isRequired
};

export {CareerContainer as Component};
export default withTraitify(CareerContainer);
