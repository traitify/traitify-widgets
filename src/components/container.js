/* eslint-disable react/forbid-prop-types */
import PropTypes from "prop-types";
import {useEffect} from "react";
import {RecoilRoot, useRecoilState, useSetRecoilState} from "recoil";
import slice from "lib/common/object/slice";
import Http from "lib/http";
import I18n from "lib/i18n";
import Listener from "lib/listener";
import {
  assessmentIDState,
  benchmarkIDState,
  httpState,
  i18nState,
  listenerState,
  loadingState,
  localeState,
  optionsState,
  profileIDState
} from "lib/recoil";

function State({children, http: _http, i18n: _i18n, listener: _listener, options}) {
  const setAssessmentID = useSetRecoilState(assessmentIDState);
  const setBenchmarkID = useSetRecoilState(benchmarkIDState);
  const setHttp = useSetRecoilState(httpState);
  const setI18n = useSetRecoilState(i18nState);
  const setListener = useSetRecoilState(listenerState);
  const setLocale = useSetRecoilState(localeState);
  const setOptions = useSetRecoilState(optionsState);
  const setProfileID = useSetRecoilState(profileIDState);
  const [loading, setLoading] = useRecoilState(loadingState);
  const {
    assessmentID,
    benchmarkID,
    locale,
    profileID,
    ...extraOptions
  } = options;

  useEffect(() => {
    setOptions(extraOptions);
  }, [options]);

  useEffect(() => {
    if(!locale) { return; }

    setLocale(locale);
  }, [locale]);

  useEffect(() => {
    const http = _http || new Http(slice(options, ["authKey", "host", "version"]));

    setHttp(http);
  }, [_http]);

  useEffect(() => {
    const i18n = _i18n || new I18n();

    setI18n(i18n);
  }, [_i18n]);

  useEffect(() => {
    const listener = _listener || new Listener();

    setListener(listener);
  }, [_listener]);

  useEffect(() => {
    setAssessmentID(assessmentID);
  }, [assessmentID]);

  useEffect(() => {
    setBenchmarkID(benchmarkID);
  }, [benchmarkID]);

  useEffect(() => {
    setProfileID(profileID);
  }, [profileID]);

  useEffect(() => {
    setLoading(false);
  }, []);

  if(loading) { return null; }

  return children;
}

State.defaultProps = {
  http: null,
  i18n: null,
  listener: null,
  options: {}
};
State.propTypes = {
  children: PropTypes.node.isRequired,
  http: PropTypes.object,
  i18n: PropTypes.object,
  listener: PropTypes.object,
  options: PropTypes.shape({
    assessmentID: PropTypes.string,
    benchmarkID: PropTypes.string,
    locale: PropTypes.string,
    profileID: PropTypes.string
  })
};

export default function Container(props) {
  return (
    <RecoilRoot>
      <State {...props} />
    </RecoilRoot>
  );
}
