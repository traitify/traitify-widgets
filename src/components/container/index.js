import {RecoilRoot} from "recoil";
import ErrorBoundary from "./error-boundary";
import State from "./state";

export default function Container(props) {
  return (
    <RecoilRoot>
      <ErrorBoundary>
        <State {...props} />
      </ErrorBoundary>
    </RecoilRoot>
  );
}
