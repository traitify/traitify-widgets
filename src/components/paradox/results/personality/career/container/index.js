import PropTypes from "prop-types";
import {useEffect} from "react";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import CareerFilter from "../filter";
import CareerModal from "../modal";
import CareerResults from "../list";
import style from "./style.scss";

function Careers(props) {
  const {ui, isReady} = props;

  useEffect(() => {
    ui.trigger("Careers.initialized", {props});
  }, []);
  useEffect(() => {
    ui.trigger("Careers.updated", {props});
  });

  return (
    isReady("results") && (
      <div className={style.container}>
        <CareerFilter {...props} />
        <CareerResults {...props} />
        <CareerModal {...props} />
      </div>
    )
  );
}
Careers.propTypes = {
  isReady: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};
export {Careers as Component};
export default withTraitify(Careers);
