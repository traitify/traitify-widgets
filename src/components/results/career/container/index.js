import useComponentEvents from "lib/hooks/use-component-events";
import useResults from "lib/hooks/use-results";
import style from "./style.scss";
import CareerFilter from "../filter";
import CareerList from "../list";
import CareerModal from "../modal";

export default function CareerContainer() {
  const results = useResults({surveyType: "personality"});

  useComponentEvents("Careers");

  if(!results) { return null; }

  return (
    <div className={style.container}>
      <CareerFilter />
      <CareerList />
      <CareerModal />
    </div>
  );
}
