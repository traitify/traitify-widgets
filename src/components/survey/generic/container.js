import PropTypes from "prop-types";
import style from "./style.scss";

export default function Container({children, progress}) {
  return (
    <div className={`${style.container}`}>
      {progress < 100
        && (
        <div className={style.progressBar}>
          <div className={style.progress} style={{width: `${progress}%`}} />
        </div>
        )}
      {children}
    </div>
  );
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
  progress: PropTypes.number.isRequired
};
