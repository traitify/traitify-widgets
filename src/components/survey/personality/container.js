import {
  faChevronLeft,
  faMaximize,
  faMinimize
} from "@fortawesome/free-solid-svg-icons";
import PropTypes from "prop-types";
import {useEffect, useLayoutEffect, useRef} from "react";
import Icon from "components/common/icon";
import dig from "lib/common/object/dig";
import useAssessment from "lib/hooks/use-assessment";
import useElementSize from "lib/hooks/use-element-size";
import useFullscreen from "lib/hooks/use-fullscreen";
import useOption from "lib/hooks/use-option";
import useTranslate from "lib/hooks/use-translate";
import Responses from "./responses";
import style from "./style.scss";

function Container({
  caption = null,
  children,
  dispatch,
  likert = false,
  onResponse = null,
  progress,
  slideIndex
}) {
  const assessment = useAssessment({surveyType: "personality"});
  const {allowFullscreen, ...options} = useOption("survey") || {};
  const container = useRef(null);
  const content = useRef(null);
  const [fullscreen, toggleFullscreen] = useFullscreen(container.current);
  const size = useElementSize(content.current);
  const text = useRef(null);
  const translate = useTranslate();

  useEffect(() => { dispatch({size, type: "resize"}); }, [...size]);
  useLayoutEffect(() => {
    if(!text.current) { return; }

    text.current.focus({preventScroll: true});
  }, [caption]);

  const back = () => dispatch({type: "back"});
  const textSurvey = dig(assessment, "slide_type")?.toLowerCase() === "text";
  const allowBack = Object.hasOwn(options, "allowBack") ? options.allowBack : textSurvey;

  return (
    <div className={`${style.container} traitify--survey-${textSurvey ? "text" : "image"}`} ref={container}>
      {caption && !textSurvey && (
        <div className={style.caption} ref={text} tabIndex="-1">{caption}</div>
      )}
      {textSurvey && (
        <div className={style.progressBar}>
          <div className={style.progress} style={{width: `${progress}%`}} />
        </div>
      )}
      <div className={style.content} ref={content}>
        {!textSurvey && <div className={style.progress} style={{width: `${progress}%`}} />}
        {children}
        {allowBack && slideIndex > 0 && (
          <button className={style.back} onClick={back} type="button">
            <Icon alt={translate("back")} icon={faChevronLeft} />
          </button>
        )}
        {allowFullscreen && (
          <button className={style.fullscreen} onClick={toggleFullscreen} type="button">
            <Icon alt="fullscreen" icon={fullscreen ? faMinimize : faMaximize} />
          </button>
        )}
      </div>
      <Responses likert={likert} onResponse={onResponse} />
    </div>
  );
}

Container.propTypes = {
  caption: PropTypes.string,
  children: PropTypes.node.isRequired,
  dispatch: PropTypes.func.isRequired,
  likert: PropTypes.bool,
  onResponse: PropTypes.func,
  progress: PropTypes.number.isRequired,
  slideIndex: PropTypes.number.isRequired
};

export default Container;
