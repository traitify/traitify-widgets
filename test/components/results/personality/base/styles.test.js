import Component from "components/results/personality/base/styles";
import mutable from "lib/common/object/mutable";
import ComponentHandler from "support/component-handler";
import {mockAssessment} from "support/container/http";
import {mockOption} from "support/container/options";
import useContainer from "support/hooks/use-container";
import _assessment from "support/json/assessment/type-based.json";

describe("PersonalityBaseStyles", () => {
  let assessment;
  let component;

  useContainer();

  beforeEach(() => {
    assessment = mutable(_assessment);
    assessment.personality_types[0].personality_type.career_style = ["Creating", "Building from scratch", "Diving in deep"];
    assessment.personality_types[1].personality_type.career_style = ["Taking charge", "Setting trends", "Changing"];
    assessment.personality_types[2].personality_type.career_style = ["Goal-setting", "Equipment using", "Feasibility monitoring"];
    assessment.personality_types[3].personality_type.career_style = ["Studying", "Problem solving", "Mind-expanding"];
    assessment.personality_types[4].personality_type.career_style = ["Following steps", "Organizing", "Tracking details"];
    assessment.personality_types[5].personality_type.career_style = ["Helping", "Guiding", "Listening"];

    mockAssessment(assessment);
  });

  describe("callbacks", () => {
    it("triggers initialization", async() => {
      await ComponentHandler.setup(Component);

      expect(container.listener.trigger).toHaveBeenCalledWith(
        "PersonalityStyles.initialized",
        expect.objectContaining({styles: expect.any()})
      );
    });

    it("triggers update", async() => {
      component = await ComponentHandler.setup(Component);
      await component.update();

      expect(container.listener.trigger).toHaveBeenCalledWith(
        "PersonalityStyles.updated",
        expect.objectContaining({styles: expect.any()})
      );
    });
  });

  it("renders component", async() => {
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders component with headers", async() => {
    mockOption("showHeaders", true);
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders component with less types", async() => {
    assessment.personality_types = [
      assessment.personality_types[0],
      assessment.personality_types[1]
    ];
    mockAssessment(assessment);
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if disabled", async() => {
    mockOption("disabledComponents", ["PersonalityStyles"]);
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if no styles", async() => {
    assessment = mutable(_assessment);
    mockAssessment(assessment);
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if results not ready", async() => {
    mockAssessment(null);
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });
});
