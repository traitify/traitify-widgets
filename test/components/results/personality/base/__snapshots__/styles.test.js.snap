// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalityBaseStyles renders component 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
>
  <div
    className="container"
  >
    <div
      className="columns"
    >
      <div>
        <div
          className="heading"
        >
          More elements of:
        </div>
        <div
          className="styles"
        >
          <div
            className="style"
          >
            <img
              alt="Visionary"
              src="https://images.ctfassets.net/4pzvszxmf15y/lgBRClpXEDIlGRUwox17h/cddacc1c0c4df6704c7a09e2b908cd5d/visionary-black.png"
            />
            <span>
              Helping
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Visionary"
              src="https://images.ctfassets.net/4pzvszxmf15y/lgBRClpXEDIlGRUwox17h/cddacc1c0c4df6704c7a09e2b908cd5d/visionary-black.png"
            />
            <span>
              Guiding
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Visionary"
              src="https://images.ctfassets.net/4pzvszxmf15y/lgBRClpXEDIlGRUwox17h/cddacc1c0c4df6704c7a09e2b908cd5d/visionary-black.png"
            />
            <span>
              Listening
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Planner"
              src="https://images.ctfassets.net/4pzvszxmf15y/6mTdrGCzYEeErSkT96Lx4Q/77bd1dafc3700891da6a4d0e568057b0/planner-black.png"
            />
            <span>
              Following steps
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Planner"
              src="https://images.ctfassets.net/4pzvszxmf15y/6mTdrGCzYEeErSkT96Lx4Q/77bd1dafc3700891da6a4d0e568057b0/planner-black.png"
            />
            <span>
              Organizing
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Planner"
              src="https://images.ctfassets.net/4pzvszxmf15y/6mTdrGCzYEeErSkT96Lx4Q/77bd1dafc3700891da6a4d0e568057b0/planner-black.png"
            />
            <span>
              Tracking details
            </span>
          </div>
        </div>
      </div>
      <div
        className="negative"
      >
        <div
          className="heading"
        >
          Not so much of:
        </div>
        <div
          className="styles"
        >
          <div
            className="style"
          >
            <img
              alt="Naturalist"
              src="https://images.ctfassets.net/4pzvszxmf15y/afwpXsCa69nCw6EUu3ecr/6eba34e8d15e9974926336eaf8356709/naturalist-black.png"
            />
            <span>
              Creating
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Naturalist"
              src="https://images.ctfassets.net/4pzvszxmf15y/afwpXsCa69nCw6EUu3ecr/6eba34e8d15e9974926336eaf8356709/naturalist-black.png"
            />
            <span>
              Building from scratch
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Naturalist"
              src="https://images.ctfassets.net/4pzvszxmf15y/afwpXsCa69nCw6EUu3ecr/6eba34e8d15e9974926336eaf8356709/naturalist-black.png"
            />
            <span>
              Diving in deep
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Analyzer"
              src="https://images.ctfassets.net/4pzvszxmf15y/1MUfph7kuAypJM6bZEm4c4/4699694511bb00c1b6e78e74917629e5/analyzer-black.png"
            />
            <span>
              Taking charge
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Analyzer"
              src="https://images.ctfassets.net/4pzvszxmf15y/1MUfph7kuAypJM6bZEm4c4/4699694511bb00c1b6e78e74917629e5/analyzer-black.png"
            />
            <span>
              Setting trends
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Analyzer"
              src="https://images.ctfassets.net/4pzvszxmf15y/1MUfph7kuAypJM6bZEm4c4/4699694511bb00c1b6e78e74917629e5/analyzer-black.png"
            />
            <span>
              Changing
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PersonalityBaseStyles renders component with headers 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
>
  <div
    className="container"
  >
    <div
      className="sectionHeading"
    >
      Your overall career style is...
    </div>
    <div
      className="columns"
    >
      <div>
        <div
          className="heading"
        >
          More elements of:
        </div>
        <div
          className="styles"
        >
          <div
            className="style"
          >
            <img
              alt="Visionary"
              src="https://images.ctfassets.net/4pzvszxmf15y/lgBRClpXEDIlGRUwox17h/cddacc1c0c4df6704c7a09e2b908cd5d/visionary-black.png"
            />
            <span>
              Helping
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Visionary"
              src="https://images.ctfassets.net/4pzvszxmf15y/lgBRClpXEDIlGRUwox17h/cddacc1c0c4df6704c7a09e2b908cd5d/visionary-black.png"
            />
            <span>
              Guiding
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Visionary"
              src="https://images.ctfassets.net/4pzvszxmf15y/lgBRClpXEDIlGRUwox17h/cddacc1c0c4df6704c7a09e2b908cd5d/visionary-black.png"
            />
            <span>
              Listening
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Planner"
              src="https://images.ctfassets.net/4pzvszxmf15y/6mTdrGCzYEeErSkT96Lx4Q/77bd1dafc3700891da6a4d0e568057b0/planner-black.png"
            />
            <span>
              Following steps
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Planner"
              src="https://images.ctfassets.net/4pzvszxmf15y/6mTdrGCzYEeErSkT96Lx4Q/77bd1dafc3700891da6a4d0e568057b0/planner-black.png"
            />
            <span>
              Organizing
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Planner"
              src="https://images.ctfassets.net/4pzvszxmf15y/6mTdrGCzYEeErSkT96Lx4Q/77bd1dafc3700891da6a4d0e568057b0/planner-black.png"
            />
            <span>
              Tracking details
            </span>
          </div>
        </div>
      </div>
      <div
        className="negative"
      >
        <div
          className="heading"
        >
          Not so much of:
        </div>
        <div
          className="styles"
        >
          <div
            className="style"
          >
            <img
              alt="Naturalist"
              src="https://images.ctfassets.net/4pzvszxmf15y/afwpXsCa69nCw6EUu3ecr/6eba34e8d15e9974926336eaf8356709/naturalist-black.png"
            />
            <span>
              Creating
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Naturalist"
              src="https://images.ctfassets.net/4pzvszxmf15y/afwpXsCa69nCw6EUu3ecr/6eba34e8d15e9974926336eaf8356709/naturalist-black.png"
            />
            <span>
              Building from scratch
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Naturalist"
              src="https://images.ctfassets.net/4pzvszxmf15y/afwpXsCa69nCw6EUu3ecr/6eba34e8d15e9974926336eaf8356709/naturalist-black.png"
            />
            <span>
              Diving in deep
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Analyzer"
              src="https://images.ctfassets.net/4pzvszxmf15y/1MUfph7kuAypJM6bZEm4c4/4699694511bb00c1b6e78e74917629e5/analyzer-black.png"
            />
            <span>
              Taking charge
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Analyzer"
              src="https://images.ctfassets.net/4pzvszxmf15y/1MUfph7kuAypJM6bZEm4c4/4699694511bb00c1b6e78e74917629e5/analyzer-black.png"
            />
            <span>
              Setting trends
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Analyzer"
              src="https://images.ctfassets.net/4pzvszxmf15y/1MUfph7kuAypJM6bZEm4c4/4699694511bb00c1b6e78e74917629e5/analyzer-black.png"
            />
            <span>
              Changing
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PersonalityBaseStyles renders component with less types 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
>
  <div
    className="container"
  >
    <div
      className="columns"
    >
      <div>
        <div
          className="heading"
        >
          More elements of:
        </div>
        <div
          className="styles"
        >
          <div
            className="style"
          >
            <img
              alt="Analyzer"
              src="https://images.ctfassets.net/4pzvszxmf15y/1MUfph7kuAypJM6bZEm4c4/4699694511bb00c1b6e78e74917629e5/analyzer-black.png"
            />
            <span>
              Taking charge
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Analyzer"
              src="https://images.ctfassets.net/4pzvszxmf15y/1MUfph7kuAypJM6bZEm4c4/4699694511bb00c1b6e78e74917629e5/analyzer-black.png"
            />
            <span>
              Setting trends
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Analyzer"
              src="https://images.ctfassets.net/4pzvszxmf15y/1MUfph7kuAypJM6bZEm4c4/4699694511bb00c1b6e78e74917629e5/analyzer-black.png"
            />
            <span>
              Changing
            </span>
          </div>
        </div>
      </div>
      <div
        className="negative"
      >
        <div
          className="heading"
        >
          Not so much of:
        </div>
        <div
          className="styles"
        >
          <div
            className="style"
          >
            <img
              alt="Naturalist"
              src="https://images.ctfassets.net/4pzvszxmf15y/afwpXsCa69nCw6EUu3ecr/6eba34e8d15e9974926336eaf8356709/naturalist-black.png"
            />
            <span>
              Creating
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Naturalist"
              src="https://images.ctfassets.net/4pzvszxmf15y/afwpXsCa69nCw6EUu3ecr/6eba34e8d15e9974926336eaf8356709/naturalist-black.png"
            />
            <span>
              Building from scratch
            </span>
          </div>
          <div
            className="style"
          >
            <img
              alt="Naturalist"
              src="https://images.ctfassets.net/4pzvszxmf15y/afwpXsCa69nCw6EUu3ecr/6eba34e8d15e9974926336eaf8356709/naturalist-black.png"
            />
            <span>
              Diving in deep
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PersonalityBaseStyles renders nothing if disabled 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
/>
`;

exports[`PersonalityBaseStyles renders nothing if no styles 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
/>
`;

exports[`PersonalityBaseStyles renders nothing if results not ready 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
/>
`;
